packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -1616,15 +1620,34 @@ class _SelectableFragment with Selectable, ChangeNotifier implements TextLayoutM
     return result;
   }
 
-  TextPosition _getNextPosition(TextBoundary boundary, TextPosition position, bool forward) {
-    if (forward) {
-      return _clampTextPosition(
-        (PushTextPosition.forward + boundary).getTrailingTextBoundaryAt(position)
-      );
+  // These 2 methods are copied from editable_text.dart. Ideally the code can
+  // be shared.
```

Comment:
I guess the only thing you're using from this class is range, right? You could almost take that as a parameter and put these somewhere as static methods... I guess you might have bigger plans for organizing these in the long term, though?

Reply:
yeah, probably don't need affinity here at all. Removed the comment.